homeassistant/components/qnap_qsw/sensor.py

```diff
@@ -152,20 +157,187 @@ class QswSensorEntityDescription(SensorEntityDescription, QswEntityDescription):
+        qsw_type=QswEntityType.PORT,
+        state_class=SensorStateClass.MEASUREMENT,
+        subkey=QSD_SPEED,
+    ),
+    QswSensorEntityDescription(
+        entity_registry_enabled_default=False,
+        icon="mdi:download-network",
+        key=QSD_PORTS_STATISTICS,
+        name="RX",
+        native_unit_of_measurement=DATA_BYTES,
+        qsw_type=QswEntityType.PORT,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        subkey=QSD_RX_OCTETS,
+    ),
+    QswSensorEntityDescription(
+        entity_registry_enabled_default=False,
+        icon="mdi:close-network",
+        key=QSD_PORTS_STATISTICS,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        name="RX Errors",
+        qsw_type=QswEntityType.PORT,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        subkey=QSD_RX_ERRORS,
+    ),
+    QswSensorEntityDescription(
+        device_class=SensorDeviceClass.DATA_RATE,
+        entity_registry_enabled_default=False,
+        icon="mdi:download-network",
+        key=QSD_PORTS_STATISTICS,
+        name="RX Speed",
+        native_unit_of_measurement=UnitOfDataRate.BYTES_PER_SECOND,
+        qsw_type=QswEntityType.PORT,
+        state_class=SensorStateClass.MEASUREMENT,
+        subkey=QSD_RX_SPEED,
+    ),
+    QswSensorEntityDescription(
+        entity_registry_enabled_default=False,
+        icon="mdi:upload-network",
+        key=QSD_PORTS_STATISTICS,
+        name="TX",
+        native_unit_of_measurement=DATA_BYTES,
+        qsw_type=QswEntityType.PORT,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        subkey=QSD_TX_OCTETS,
+    ),
+    QswSensorEntityDescription(
+        device_class=SensorDeviceClass.DATA_RATE,
+        entity_registry_enabled_default=False,
+        icon="mdi:upload-network",
+        key=QSD_PORTS_STATISTICS,
+        name="TX Speed",
+        native_unit_of_measurement=UnitOfDataRate.BYTES_PER_SECOND,
+        qsw_type=QswEntityType.PORT,
+        state_class=SensorStateClass.MEASUREMENT,
+        subkey=QSD_TX_SPEED,
+    ),
+)
+
 
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Add QNAP QSW sensors from a config_entry."""
     coordinator: QswDataCoordinator = hass.data[DOMAIN][entry.entry_id][QSW_COORD_DATA]
-    async_add_entities(
-        QswSensor(coordinator, description, entry)
-        for description in SENSOR_TYPES
+
+    entities: list[QswSensor] = []
+
+    for description in SENSOR_TYPES:
         if (
             description.key in coordinator.data
             and description.subkey in coordinator.data[description.key]
-        )
-    )
+        ):
+            entities.append(QswSensor(coordinator, description, entry))
+
+    for description in LACP_PORT_SENSOR_TYPES:
+        if (
+            description.key in coordinator.data
+            and QSD_LACP_PORTS in coordinator.data[description.key]
+        ):
```

Comment:
I'd reverse this conditional and continue here and below to reduce the indentation level.

Reply:
Done in ee03a0308519d6f7c7b3765b99288dcf642fa22d